### PR TITLE
SRE-57-add-index-resource

### PR DESCRIPTION
**Pull request by maxd-wttj**
Addresses #11

### CODE REVIEW

File: internal/provider/index_resource_test.go [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+				},
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestCheckResourceAttr("meilisearch_index.test", "uid", "abcdefg"),
+				),
+			},
+			{
+				ResourceName:            "meilisearch_index.test",
+				ImportStateId:           "abcdefg",
+				ImportState:             true,
+				ImportStateVerify:       false,
```
Comment (pauldn-wttj) [inline internal/provider/index_resource_test.go:81]:
```suggestion
				ImportStateVerify:       true,
```

Is there a reason not to check the state values after importing the resource?

**Review by pauldn-wttj — COMMENTED** [review]
:heart_eyes:  :tada:

File: internal/provider/index_resource.go [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"primary_key": schema.StringAttribute{
+				Description: "Primary key of the index (`null` if not specified and if no documents have been added yet, see [official documentation](https://www.meilisearch.com/docs/learn/core_concepts/primary_key#meilisearch-guesses-your-primary-key) for more details).",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
```
[the remark this answers is not in the document]
Comment (maxd-wttj) [reply internal/provider/index_resource.go:65]: this is why we implemented it this way :-)

File: internal/provider/index_resource_test.go [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+			// Read testing
+			{
+
+				PreConfig: func() {
+					_, err := client.DeleteIndex("abcdef")
+					if err != nil {
+						return
+					}
+				},
+
```
[the remark this answers is not in the document]
Comment (maxd-wttj) [reply internal/provider/index_resource_test.go:29]: When running the tests several times in a row and they are failing, the volume is not cleared. This is a way to ensure the index is absent

File: internal/provider/index_resource_test.go (hunk 1 above)
Comment (maxd-wttj) [reply internal/provider/index_resource_test.go:81]: nope, we need some clarification here

File: internal/provider/index_resource_test.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (pauldn-wttj) [reply internal/provider/index_resource_test.go:29]: I'm not sure it's the Go test responsibility to make sure the test environment is not corrupted.
If we can have this kind of issue in these tests, I'd prefer we tackle this once and for all by ensuring the `testacc` target in our `GNUMakefile` calls the `clean` dependency:

https://github.com/pauldn-wttj/terraform-provider-meilisearch/blob/df789d29d5b2da7d77dac745adfea307134aa11f/GNUmakefile#L6

=> 

```Makefile
testacc: clean
```

This way we don't have to do that individually in each file, what do you think?

File: internal/provider/index_resource.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (pauldn-wttj) [reply internal/provider/index_resource.go:65]: https://github.com/pauldn-wttj/terraform-provider-meilisearch/issues/119

[the remark this answers is not in the document]
Comment (pauldn-wttj) [reply internal/provider/index_resource.go:65]: The implementation actually exists if fetching the existing index, I misread that. The code is:

```go
client.Index("movies").UpdateIndex("id")
```

However, it's not mandatory for a v1, I'd say.

**Pull request merged**